tests/testthat/test-as-data-frame.R

```diff
@@ -76,6 +76,63 @@ with_mock_crunch({
         expect_identical(evalq(gender, as.data.frame(ds)),
             as.vector(ds$gender))
     })
+    csv_df <- read.csv("dataset-fixtures/test_ds.csv", stringsAsFactors = FALSE, colClasses = "character")
+    test_that("csvToDataFrame produces the correct data frame", {
+        expected <- readRDS("dataset-fixtures/test_ds.rds")
+        cdf <- as.data.frame(ds[, c("birthyr", "gender", "location", "mymrset", "textVar", "starttime")])
+        #test local CDF variables
+        cdf$newvar <- expected$newvar <- 1:25
+        expect_identical(csvToDataFrame(csv_df, cdf), expected)
+    })
+
+    test_that("csvToDataFrame handles hidden variables", {
+        new_ds <- loadDataset("test ds")[, c("birthyr", "gender", "location", "mymrset", "textVar", "starttime")]
+        new_ds$birthyr@tuple[["discarded"]] <- TRUE
+        new_ds_df <- as.data.frame(new_ds)
+        expect_silent(
+            expect_equal(names(csvToDataFrame(csv_df, new_ds_df)),
+                c("gender", "location", "subvar2", "subvar1", "subvar3", "textVar",
+                    "starttime"))
+        )
+        # now we want the hidden vars to be included
+        new_ds_df <- as.data.frame(new_ds, include.hidden = TRUE)
+        expect_warning(
+            expect_equal(names(csvToDataFrame(csv_df, new_ds_df)),
+                c("birthyr", "gender", "location", "subvar2", "subvar1", "subvar3", "textVar",
+                    "starttime")),
+            paste0("Variable birthyr is hidden"))
+    })
+
+
+
+    test_that("coerceVariable produces expected results", {
+        df <- lapply(ds, as.vector)
+        names(df) <- names(ds)
+        df[] <- lapply(df, as.character)
+        cdf <- as.data.frame(ds)
+        df <- structure(df, class = "data.frame", row.names=c(NA, -nrow(ds)))
+        expect_is(coerceVariable(ds$mymrset, df, cdf)[[1]], "factor")
+        expect_is(coerceVariable(ds$location, df, cdf)[[1]], "factor")
+        expect_is(coerceVariable(ds$birthyr, df, cdf)[[1]], "numeric")
+        expect_is(coerceVariable(ds$textVar, df, cdf)[[1]], "character")
+        expect_is(coerceVariable(ds$starttime, df, cdf)[[1]], "Date")
+        expect_is(coerceVariable(ds$catarray, df, cdf)[[1]], "factor")
```

Comment:
It would be good to also assert the shape of these variables. It took me a bit to realize that array like variables have a different shape. It would be good to make that explicit so it's a bit easier to see.